### PR TITLE
Release

### PR DESCRIPTION
**Pull request by volodymyrtaras**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,14 @@
 
 *.txt
 
-*/out/
```
Comment (Anderwerty) [inline .gitignore]: image
What is it?

File: .gitignore [hunk 2]
```diff
@@ -4,6 +4,14 @@
...
+
+/one/
+
+/two/
+
+/three/
+
+/four/
 
 .idea/
```
Comment (Anderwerty) [inline .gitignore]: Add a blank line.

File: src/lesson/five/LambdaCalculator.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        operations.put('-', (x, y) -> x - y);
+        operations.put('*', (x, y) -> x * y);
+        operations.put('/', (x, y) -> x / y);
+        operations.put('#', (x, y) -> Math.pow(x, 1.0 / y));
+        operations.put('^', Math::pow);
+        operations.put('$', (x, y) -> Math.pow((x + y) / x + 117, y));
+        return operations;
+    }
+
+    public double calculate(double x, char c, double y) {
```
Comment (Anderwerty) [inline src/lesson/five/LambdaCalculator.java]: It is better to rename parameters.` c->operation`

File: src/lesson/five/LambdaCalculator.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        BinaryOperator<Double> action = operations.get(c);
```
Comment (Anderwerty) [inline src/lesson/five/LambdaCalculator.java]: What if you don't have such an operation? In this case better to throw a custom exception.

File: .gitignore (hunk 1 above)
Comment (volodymyrtaras) [reply .gitignore]: @Anderwerty names of unusful packages

**Comment by volodymyrtaras** [comment]
@Anderwerty fixed.

File: .gitignore (hunk 1 above)
Comment (Anderwerty) [reply .gitignore]: You should all time change your gitignore file! Please fix it.

Comment (volodymyrtaras) [reply .gitignore]: @Anderwerty fixed.

File: src/lesson/five/Start.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        System.out.println("Addition: " + lambdaCalculator.calculate(1.0, '+', 2.0));
+        System.out.println("Subtraction: " + lambdaCalculator.calculate(1.0, '-', 2.0));
+        System.out.println("Multiplication: " + lambdaCalculator.calculate(1.0, '*', 2.0));
+        System.out.println("Division: " + lambdaCalculator.calculate(1.0, '/', 2.0));
+        System.out.println("Root definition: " + lambdaCalculator.calculate(1.0, '#', 2.0));
+        System.out.println("Bringing to degree: " + lambdaCalculator.calculate(1.0, '^', 2.0));
+        System.out.println("Modified action: " + lambdaCalculator.calculate(1.0, '$', 2.0));
+    }
+}
+
```
Comment (Anderwerty) [inline src/lesson/five/Start.java]: Remove redundant blank line.

**Comment by volodymyrtaras** [comment]
@Anderwerty thank you) are there some other solutions of this task?

**Comment by Anderwerty** [comment]
Patterns could be used.

**Pull request closed without merging**